File: Wizard Battle Web API/DTOs/Spell/StaticSpellResponse.cs
﻿namespace Wizard_Battle_Web_API.DTOs.Spell
{
	public class StaticSpellResponse
	{
		public int SpellID { get; set; } = 0;

		public string SpellName { get; set; } = string.Empty;

		public string SpellDescription { get; set; } = string.Empty;

		public IconResponse Icon { get; set; } = null!;
		
		public int SpellTypeID { get; set; } = 0;

		public int SchoolCategoryID { get; set; } = 0;
		
		public decimal? DamageAmount { get; set; } = 0;

		public decimal ManaCost { get; set; } = 0;

		public decimal LifeTime { get; set; } = 0;

		public decimal CastTime { get; set; } = 0;
	}
}

File: Wizard Battle Web Client/src/app/_models/Spell/StaticSpellResponse.ts
import { IconResponse } from "../Icon/"

export interface StaticSpellResponse {
    spellID: number;
    spellName: string;
    spellDescription: string;
    icon: IconResponse;
    spellTypeID: number;
    schoolCategoryID: number;
    damageAmount: number | null;
    manaCost: number;
    lifeTime: number;
    castTime: number;
}
